Source/DafnyCore/AST/Grammar/ParserNonGeneratedPart.cs

```diff
@@ -644,6 +690,16 @@ private ModuleKindEnum GetModuleKind(DeclModifierData mods) {
     return ModuleKindEnum.Concrete;
   }
 
+  /// <summary>
+  /// Before literals that end a block, we usually add CheckNoAttributes to avoid any non-attached or dangling attributes
+  /// </summary>
+  public void CheckNoAttributes(ref Attributes attrs) {
+    if (attrs != null) {
+      SemErr(ErrorId.p_extra_attributes, attrs.RangeToken, "Attribute not expected here");
```

Comment:
Needs a test case or two

Reply:
I can't test it for now because the way parsing code is generated, it's unreachable, but on the next update of the grammar there will be cases and this will be tested.